qiskit/compiler/assemble.py

```diff
@@ -298,7 +299,16 @@ def _parse_pulse_args(backend, qubit_lo_freq, meas_lo_freq, qubit_lo_range,
 
     if dynamic_reprate_enabled:
         rep_delay = rep_delay or getattr(backend_config, 'default_rep_delay', None)
-        rep_delay = rep_delay * 1e6  # convert sec to μs
+        if rep_delay is not None:
+            rep_delay_range = getattr(backend_config, 'rep_delay_range', None)
+            if rep_delay_range is None or len(rep_delay_range) != 2:
```

Comment:
If rep_delay_range is None `len(rep_delay_range)` will raise an error.

Reply:
It shouldn't check the second statement if `rep_delay_range=None` though?

Reply:
Updated this to be simpler, addressed in https://github.com/Qiskit/qiskit-terra/pull/4830/commits/c9f0add598771de9a269a06f1de05f44abd89903